### ISSUE TITLE
Add fuzzy search to narrow down snippets in tree view

### ISSUE THREAD

**Issue by ecool** [RandomFractals/vscode-snippets-viewer#16]
Allow for the ability to fuzzy search for snippets with the ability to select which language. 
Could also have the selected language be selected based on active editor language id.

Possible question about what happens when an active editor language embeds other languages into it. ie: HTML which allows for embedding CSS or JavaScript.

**Comment by RandomFractals** [comment]
@ecool I think #18 and #19 should cover limiting display of all built-in snippets and snippets for certain languages to some extent.

By fuzzy search, do you mean adding Search Snippets Input and display results for matching snippet name with regex?

As for language snippets auto-select, I'll just show snippets listed for the active editor document language id in #10. Users should be able to navigate to other language snippets if desired. Some actually list other language snippets already.

**Comment by ecool** [comment]
@RandomFractals I meant with language id being used in limiting which snippets are shown in the search.

Input box for searching based on regex matching would work and can be narrowed further by selecting a language to search through which could be a dropdown.

However, it could be combined with #10 so only need to type in your search.

**Comment by RandomFractals** [comment]
well, you can select a language from a tree. I think that should be the same for search results too, unless explicitly selected to skip language in #19.

I'll add Search Snippets input to show all matching snippets. Not sure how useful it will be, but maybe that's one way to see `import` snippets in all languages :)

**Comment by ecool** [comment]
Opening up the specific languages in the tree to search would work like what I was thinking.

**Comment by RandomFractals** [comment]
I can add Search button to root and language nodes. Then Search invoked on snippet language node would prompt for a regex to match only for snippets under that language. Would that work?

Bigger questions are:

1. Do we show results in the snippets webview I plan to add in #5?
2. or display results in snippets tree view and have another button on clear search?
3. Do we want to save entered search patterns and let users enter new search or pick one of the recently used?

This can get complicated. Still not sure how useful searching snippets for `import` or `component` across all snippet files for a language would be.

**Comment by ecool** [comment]
Hmm, I was thinking of something similar as to how the Command Pallete behaves and just double checking now shows that it already has a way to filter snippets. `Insert Snippet` allows for filtering and searching of snippets affecting the active editor. This may not be needed.

If it was possible to show a tooltip of the snippet that is currently selected in the Command Pallete that would be great but not sure if that behavior fits into this extension.

**Comment by RandomFractals** [comment]
ah, yes. I forgot about that Insert Snippets command feature, even though it's listed in vscode docs: https://code.visualstudio.com/docs/editor/userdefinedsnippets#_builtin-snippets

Let's still keep it open since that build-in snippets search option doesn't show snippet body and full info. So, I still think you idea is good and we just need to decide if we show search results in a tree, or a table view I'll be adding, or both. Then you'll be able to see all the snippet info.

I can make that Search prompt behave similar to Input Snippet that shows matching snippets as you type name or description filter.

**Comment by ecool** [comment]
Yeah, Insert Snippet would work much better if it had a tooltip popup showing the snippet.
